src/components/patterns/incrementer/Incrementer.js

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import { PropTypes } from 'prop-types';
+import { noop } from 'lodash';
+import styled from 'styled-components';
+
+import Icon from '../../base/icons/Icon';
+import Button from '../../controls/buttons/Button';
+import { InputBase } from '../../controls/BaseControls';
+import getValidationStateVariables
+  from '../../controls/getValidationStateVariables';
+
+const IncrementerTextbox = styled(InputBase)`
+  margin: 0 10px;
+  text-align: center;
+  width: 60px;
+`;
+
+class Incrementer extends React.Component {
+  static propTypes = {
+    /** The value to start the incrementer at */
+    startingValue: PropTypes.number,
+    /** The amount to increment the value by */
+    incrementAmount: PropTypes.number,
+    /** The amount to decrement the value by */
+    decrementAmount: PropTypes.number,
+    /** The highest value the incrementer can be incremented to */
+    upperThreshold: PropTypes.number,
+    /** The lowest value the incrementer can be decremented to */
+    lowerThreshold: PropTypes.number,
+    /** Function to execute with the new value */
+    onValueUpdated: PropTypes.func
+  };
+
+  static defaultProps = {
+    startingValue: 0,
+    incrementAmount: 1,
+    decrementAmount: 1,
+    onValueUpdated: noop
+  };
+
+  constructor(props) {
+    super(props);
+
+    const { startingValue } = this.props;
+
+    this.decrementValue = this.decrementValue.bind(this);
+    this.incrementValue = this.incrementValue.bind(this);
```

Comment:
Eh, it's just habit for me. There's no real argument for/against that is strong enough for me to favor either. 🤷‍♂️